proto/minder/v1/minder.proto

```diff
@@ -1139,7 +1139,15 @@ message DiffType {
         // depfile is the file that contains the dependencies for this ecosystem
         string depfile = 2;
     }
+
+    // ecosystems is the list of ecosystems to be used
+    // for the "dep" diff type.
     repeated Ecosystem ecosystems = 1;
+
+    // type is the type of diff ingestor to use.
+    // The default is "dep" which will leverage
+    // the ecosystems array.
+    string type = 2;
```

Comment:
This seems fine -- I might have suggested using a `oneof`, but I seem to recall that those can actually be a bit a bit tricky in proto (plus we'd have the issue of doing a compatibility bridge dance).

Reply:
The main thing preventing us from using `oneof` (which would have been handy for many cases) is stuff like these:
* https://github.com/stacklok/minder/blob/main/pkg/api/protobuf/go/minder/v1/rule_types.go#L30-L37
* https://github.com/stacklok/minder/blob/main/internal/util/jsonyaml/jsonyamlutils.go#L52-L60

changing those to use `protojson` instead of `json`, while it would do the trick to correctly parse protobuf as we'd need.. For some reason it also currently breaks parsing either rule types or profiles. So we haven't changed it because of that.